### PR TITLE
add vault secret decoder

### PR DESCRIPTION
**Pull request by svenwltr**
Wdyt?

Replaces this noisy code: https://github.com/rebuy-de/boombot/blob/main/pkg/dal/vault/vault.go#L74-L95

See https://github.com/rebuy-de/kubot/pull/304

### CODE REVIEW

File: pkg/vaultutil/secrets.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+type SlackSecrets struct {
+	VerificationToken string `vault:"slack-verification-token"`
+	SigningSecret     string `vault:"slack-signing-secret"`
+	Token             string `vault:"slack-token"`
+	AppToken          string `vault:"slack-app-token"`
+	Channel           string `vault:"slack-channel"`
+}
+
+type OIDCSecrets struct {
+	ClientID     string `vault:"oidc-client-id"`
```
Comment (svenwltr) [inline pkg/vaultutil/secrets.go:52]: The original struct in cluster-config uses `_` (ie `oidc_client_id`). I want to have the same separator everywhere, but I do not care which one.

**Review by bjoernhaeuser — APPROVED** [review]

File: pkg/vaultutil/secrets.go (hunk 1 above)
Comment (der-eismann) [reply pkg/vaultutil/secrets.go:52]: I used dashes for boombot already, no hard feeling as well.

**Review by der-eismann — APPROVED** [review]

**Comment by svenwltr** [comment]
I updated the decoder and am using the one that I already tested in keydb-inspector (one which actually works in production). Also I removed the struct presets, because I think they do not fit here. We can add them later, if we think it makes sense.

**Review by bjoernhaeuser — APPROVED** [review]

**Comment by svenwltr** [comment]
bump

**Pull request merged**